src/api.c

```diff
@@ -301,10 +301,18 @@ API int seccomp_reset(scmp_filter_ctx ctx, uint32_t def_action)
 {
 	struct db_filter_col *col = (struct db_filter_col *)ctx;
 
-	/* use a NULL filter collection here since we are resetting it */
-	if (ctx == NULL || db_col_action_valid(NULL, def_action) < 0)
+	/* a NULL filter context indicates we are resetting the global state */
+	if (ctx == NULL) {
```

Comment:
That's a good point.  There was a `seccomp_reset(...)` test in "tests/11-basic-basic_errors.c" but I removed it because it was treating `seccomp_reset(NULL, ...)` as an error (which was correct prior to this patch).  At the very least we should restore that test with the tweak that `seccomp_reset(NULL, ...)` succeed.  I'm thinking something like this:

```diff
diff --git a/tests/11-basic-basic_errors.c b/tests/11-basic-basic_errors.c
index 3a30f412..da059df2 100644
--- a/tests/11-basic-basic_errors.c
+++ b/tests/11-basic-basic_errors.c
@@ -41,6 +41,11 @@ int main(int argc, char *argv[])
        seccomp_release(ctx);
        ctx = NULL;
 
+       /* ensure that seccomp_reset(NULL, ...) is accepted */
+       rc = seccomp_reset(NULL, SCMP_ACT_ALLOW);
+       if (rc != 0)
+               return -1;
+
        /* seccomp_load error */
        rc = seccomp_load(ctx);
        if (rc != -EINVAL)
```

Reply:
I like it.